lib/ansible/modules/cloud/vmware/vmware_dvs_portgroup.py

```diff
@@ -257,8 +274,17 @@ def create_port_group(self):
         config.defaultPortConfig = vim.dvs.VmwareDistributedVirtualSwitch.VmwarePortConfigPolicy()
         if self.module.params['vlan_trunk']:
             config.defaultPortConfig.vlan = vim.dvs.VmwareDistributedVirtualSwitch.TrunkVlanSpec()
-            vlan_id_start, vlan_id_end = self.module.params['vlan_id'].split('-')
-            config.defaultPortConfig.vlan.vlanId = [vim.NumericRange(start=int(vlan_id_start.strip()), end=int(vlan_id_end.strip()))]
+            vlan_id_list = []
+            for vlan_id_splitted in self.module.params['vlan_id'].split(','):
+                try:
+                    vlan_id_start, vlan_id_end = vlan_id_splitted.split('-')
+                    vlan_id_list.append(vim.NumericRange(start=int(vlan_id_start.strip()), end=int(vlan_id_end.strip())))
+                except ValueError:
+                    vlan_id_list.append(vim.NumericRange(start=int(vlan_id_splitted.strip()), end=int(vlan_id_splitted.strip())))
+            config.defaultPortConfig.vlan.vlanId = vlan_id_list
```

Comment:
Did you just copy the code from another PR https://github.com/ansible/ansible/pull/46628?

Reply:
Yes, this is also stated in the additional info. I have a use case where I need to have PVLAN dPG and  vlan trunk with the comma separate list, so for me it was easier to include also that 46628. If neeted I can just remove changes for this part.

Reply:
Ok. Maybe you can update your branch? The VLAN trunk change was approved. Would be better to have only the PVLAN changes in this PR.